go/libraries/utils/jwtauth/jwks.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jwtauth
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"time"
+
+	jose "gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/json"
+
+	"github.com/pquerna/cachecontrol"
+)
+
+type cachedJWKS struct {
+	value   *jose.JSONWebKeySet
+	expires time.Time
+	mutex   *sync.Mutex
+}
+
+func newCachedJWKS() *cachedJWKS {
+	return &cachedJWKS{value: nil, expires: time.Now(), mutex: &sync.Mutex{}}
+}
+
+type fetchedJWKS struct {
+	URL          string
+	BrokenBase64 bool
```

Comment:
I don't think this is necessary here...it's a kludge in the code on the infra side to account for a bug in a particular JWKS which is published by a particular AWS system.